After updating whatwg-url per https://github.com/whatwg/url/pull/156/files the following tests additionally fail:

```
  1) Web platform tests setters pathname <view-source+http://example.net/home?lang=fr#nav>.pathname = "\a\%2E\b\%2e.\c" \ is *not* a segment delimiter for non-'special' URLs:

      AssertionError: 'view-source+http://example.net/\\a\\%2E\\b\\%2e.\\c?lang=fr#nav' == 'view-source+http://example.net/\\a\\.\\b\\..\\c?lang=fr#nav'
      + expected - actual

      -view-source+http://example.net/\a\%2E\b\%2e.\c?lang=fr#nav
      +view-source+http://example.net/\a\.\b\..\c?lang=fr#nav
```

```
  2) Web platform tests setters pathname <http://example.net>.pathname = "%2e%2E%c3%89té" Bytes already percent-encoded are left as-is, except %2E.:

      AssertionError: 'http://example.net/%2e%2E%c3%89t%C3%A9' == 'http://example.net/..%c3%89t%C3%A9'
      + expected - actual

      -http://example.net/%2e%2E%c3%89t%C3%A9
      +http://example.net/..%c3%89t%C3%A9
```

--- a/url/setters_tests.json
+++ b/url/setters_tests.json
@@ -952,8 +952,8 @@
             "href": "view-source+http://example.net/home?lang=fr#nav",
             "new_value": "\\a\\%2E\\b\\%2e.\\c",
             "expected": {
-                "href": "view-source+http://example.net/\\a\\.\\b\\..\\c?lang=fr#nav",
-                "pathname": "/\\a\\.\\b\\..\\c"
+                "href": "view-source+http://example.net/\\a\\%2E\\b\\%2e.\\c?lang=fr#nav",
+                "pathname": "/\\a\\%2E\\b\\%2e.\\c"
             }
         },
         {
@@ -966,12 +966,12 @@
             }
         },
         {
-            "comment": "Bytes already percent-encoded are left as-is, except %2E.",
+            "comment": "Bytes already percent-encoded are left as-is, including %2E outside dotted segments.",
             "href": "http://example.net",
             "new_value": "%2e%2E%c3%89té",
             "expected": {
-                "href": "http://example.net/..%c3%89t%C3%A9",
-                "pathname": "/..%c3%89t%C3%A9"
+                "href": "http://example.net/%2e%2E%c3%89t%C3%A9",
+                "pathname": "/%2e%2E%c3%89t%C3%A9"
             }
         },
         {

